CatalogReaderExample/NuGet.Protocol.Catalog/CatalogClient.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace NuGet.Protocol.Catalog
+{
+    public class CatalogClient : ICatalogClient
+    {
+        private static readonly JsonSerializer JsonSerializer = CatalogJsonSerialization.Serializer;
+        private readonly HttpClient _httpClient;
+        private readonly ILogger<CatalogClient> _logger;
+
+        public CatalogClient(HttpClient httpClient, ILogger<CatalogClient> logger)
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public Task<CatalogIndex> GetIndexAsync(string indexUrl)
+        {
+            return DeserializeUrlAsync<CatalogIndex>(indexUrl);
+        }
+
+        public Task<CatalogPage> GetPageAsync(string pageUrl)
+        {
+            return DeserializeUrlAsync<CatalogPage>(pageUrl);
+        }
+
+        public async Task<CatalogLeaf> GetLeafAsync(string leafUrl)
+        {
+            // Buffer all of the JSON so we can parse twice. Once to determine the leaf type and once to deserialize
+            // the entire thing to the proper leaf type.
+            _logger.LogDebug($"Downloading {leafUrl} as a byte array.");
```

Comment:
Don't format the string yourself, let the logger do it.

Reply:
Ah, good catch. I keep switching between NuGet client's logger and Microsoft.Extensions.Logging... messes with my head.

Reply:
Fixed.